src/Nethermind/Nethermind.Network.Contract/Config/INetworkConfig.cs

```diff
@@ -103,4 +103,7 @@ public interface INetworkConfig : IConfig
 
     [ConfigItem(DefaultValue = "false", HiddenFromDocs = true, Description = "[TECHNICAL] Shutdown timeout when closing TCP port.")]
     long RlpxHostShutdownCloseTimeoutMs { get; set; }
+
+    [ConfigItem(DefaultValue = null, Description = "A comma-separated list of the part names of the client id to be masked.")]
+    string? ClientIdHiddenParts { get; set; }
```

Comment:
It needs some more information how the format looks like and what parts are there.

Reply:
Other idea is instead of providing list of parts to hide, provide a template string, that the parts would be interpolated to. Then use our standard format as default value. I think it would be more flexible.

Reply:
Good idea, I've simplified the implementation to effectively provide a public client id template. It is pretty permissive but if allowing users to put anything as the public client id is not an issue i can keep it as is.